Update GitHub Action for MarkBind (#1839)
MarkBind/markbind: docs/devGuide/githubActions/markbindAction.md

{% set title = "GitHub Actions: markbind-action" %}
<span id="title" class="d-none">{{ title }}</span>

<frontmatter>
  title: "{{ title }}"
  layout: devGuide.md
  pageNav: default
</frontmatter>

# {{ title }}

<div class="lead">

> A GitHub Action that builds and deploys your MarkBind site. The source code is at [MarkBind/markbind-action](https://github.com/MarkBind/markbind-action).

Reviewer: Maybe can add a brief explanation on the purpose of this repo?

> A GitHub Action that builds and deploys your MarkBind site. The source code is at [MarkBind/markbind-action](https://github.com/MarkBind/markbind-action).

Reviewer: Maybe can add a brief explanation on the purpose of this repo? (help users to streamline their workflow, easier for them to deploy without having to write their own actions from scratch, make their workflows consistent etc)

</div>

**Follow these steps to get started on development**

1. Fork the action repository and create a PR branch
2. Learn the workflow syntax in the repository's readme
3. Make necessary changes to the workflow files & update the readme

> 4. Set up a test repository that contains a markbind site

Reviewer: Minor nit

```suggestion
4. Set up a test repository that contains a MarkBind site
```

> 4. Set up a test repository that contains a markbind site

Reviewer: Minor nit

```suggestion
4. Set up a test repository that contains a MarkBind site
```
```suggestion
4. Set up a test repository that contains a markbind site
```

   1. Create a repository
   2. Set up MarkBind and run `markbind init` to populate with default content
   3. Push your repository to GitHub

>    4. [Example](https://github.com/tlylt/mb-test)

Reviewer 2: Resolving this since it's outdated now!

5. Include a workflow file in the repository that uses your forked version of the action
   1. Instead of using `uses: MarkBind/markbind-action@v2` in the workflow file, use `uses: yourGitHubName/markbind-action@yourBranch`

>       1. e.g. `uses: tlylt/markbind-action@master`

Reviewer: I think a bullet point will be more suitable here :)

>       1. e.g. `uses: tlylt/markbind-action@master`

Reviewer: I think a bullet point will be more suitable here :)

{% set title = "GitHub Actions: markbind-action" %}
<span id="title" class="d-none">{{ title }}</span>

<frontmatter>
  title: "{{ title }}"
  layout: devGuide.md
  pageNav: default
</frontmatter>

# {{ title }}

<div class="lead">

A GitHub Action that builds and deploys your MarkBind site. It helps users to streamline their workflow by:
* Making it easier to deploy without having to write their own actions from scratch
* Making the workflows consistent and up to date with any changes specific to MarkBind

The source code is at [MarkBind/markbind-action](https://github.com/MarkBind/markbind-action).
</div>

> **Follow these steps to get started on development**

Reviewer 2: Was a bit confused on what "development" refers to here - e.g. Are these steps for modifying MarkBind's `markbind-action`, testing `markbind-action` with a local copy, or for creating Github actions for MarkBind in general?

Reviewer 2: Maybe it would be clearer to split these steps up into different sections (I'm confused about what each step should achieve, and where the steps are leading to in general) - e.g. "Setting up", "Testing/modifying the action".

1. Fork the action repository and create a PR branch
1. Learn the workflow syntax in the repository's readme
1. Make necessary changes to the workflow files & update the readme
1. Set up a test repository that contains a MarkBind site
   1. Create a repository
   1. Set up MarkBind and run `markbind init` to populate with default content
   1. Push your repository to GitHub
      * [Example](https://github.com/MarkBind/init-typical)
1. Include a workflow file in the repository that uses your forked version of the action
   * Instead of using `uses: MarkBind/markbind-action@v2` in the workflow file, use `uses: yourGitHubName/markbind-action@yourBranch`
      * e.g. `uses: tlylt/markbind-action@master`
   * [Example](https://github.com/tlylt/mb-test/tree/main/.github/workflows)

> 1. Trigger the action as required, check and validate the run result in action logs
> 1. Submit PRs

Reviewer 2: Was initially confused if PRs were part of the development process (e.g. deploying a PR preview using `markbind-action`). I think it could be clearer to separate this from the set-up steps? e.g.

```suggestion
1. Trigger the action as required, check and validate the run result in action logs

You can now submit PRs to improve MarkBind's Github actions! 🎉 
```

{% set title = "GitHub Actions: markbind-action" %}
<span id="title" class="d-none">{{ title }}</span>

<frontmatter>
  title: "{{ title }}"
  layout: devGuide.md
  pageNav: default
</frontmatter>

# {{ title }}

<div class="lead">

A GitHub Action that builds and deploys a MarkBind site. It helps users to streamline their workflow by:
* Making it easier to deploy without having to write their own actions from scratch
* Making the workflows consistent and up to date with any changes specific to MarkBind

The source code is at [MarkBind/markbind-action](https://github.com/MarkBind/markbind-action).
</div>

**Follow these steps to get started on modifying the source code**
## Setting Up
1. Fork the [action repository](https://github.com/MarkBind/markbind-action)
1. Learn the existing workflow syntax/options provided in the repository's [readme.md](https://github.com/MarkBind/markbind-action#readme)

> 1. Set up a repository that contains a MarkBind site to test the action as a user

Reviewer 3: ```suggestion
1. Set up a repository that contains a MarkBind site to test the action:
```
(may be confusing - user of the action or MarkBind - might be best to leave it as is since it is implied (developer guide))

   1. Create a repository
   1. Set up MarkBind and run `markbind init` to populate with default content
   1. Push your repository to GitHub
      * [Example](https://github.com/MarkBind/init-typical)

## Modifying the action
1. Create a new branch from master to work on your changes
1. Make necessary modifications to the workflow files

> 1. Update the repository's readme.md to reflect the changes

Reviewer 3: ```suggestion
1. Update the repository's README.md to reflect the changes
```

## Testing the action
1. Include a workflow file in your <tooltip content="Which includes a sample MarkBind site">test repository</tooltip> to test your modified version of the action
   * Instead of using `uses: MarkBind/markbind-action@v2` in the workflow file, use `uses: yourGitHubName/markbind-action@yourBranch` to reference the unpublished version of the action that you are currently developing
      * e.g. `uses: tlylt/markbind-action@main`
   * [Example](https://github.com/tlylt/mb-test/tree/main/.github/workflows)

> 1.  Trigger the action as needed, check and validate the run result in the action logs

Reviewer 3: ```suggestion
1.  Trigger the action as needed, check and validate the results in the action logs
```
